server/src/main/java/org/jboss/as/server/Main.java

```diff
@@ -212,6 +212,10 @@ public static ServerEnvironmentWrapper determineEnvironment(String[] args, Prope
                         value = arg.substring(idx + 1);
                     }
                     systemProperties.setProperty(name, value);
+                    if (ServerEnvironment.HOME_DIR.equals(name)) {
+                        // Re-create using updated properties
+                        productConfig = ProductConfig.fromFilesystemSlot(Module.getBootModuleLoader(), WildFlySecurityManager.getPropertyPrivileged(ServerEnvironment.HOME_DIR, null), systemProperties);
+                    }
```

Comment:
I think the `WildFlySecurityManager.getPropertyPrivileged` usage here would be unnecessary, indeed it could be just replaced with `value` at this point and I don't see the need to read it again from System.properties using a privileged block.